From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 12 Feb 2020 11:16:57 -0800
Subject: [PATCH] Build: do not exit after unknown option.

---
 auto/options | 1 -
 1 file changed, 1 deletion(-)

diff --git a/auto/options b/auto/options
index 48f3a1a..1b430d4 100644
--- a/auto/options
+++ b/auto/options
@@ -403,7 +403,6 @@ $0: warning: the \"--with-sha1-asm\" option is deprecated"
 
         *)
             echo "$0: error: invalid option \"$option\""
-            exit 1
         ;;
     esac
 done
